### PR TITLE
feat(PPDSC-2314): audio player - playback speed custom children trigger

### PR DESCRIPTION
**Pull request by evgenitsn**
PPDSC-2314

**What**

Allowing users to pass custom trigger element for the playback speed control.

**I have done:**
- [x] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [x] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [x] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

New storybook examples with custom button have been added.

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2314-audio-player-playback-speed-custom-children-trigger
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2314-audio-player-playback-speed-custom-children-trigger/storybook

File: src/audio-player-composable/components/playback-speed-control/playback-speed-control.tsx [hunk 1]
```diff
@@ -63,15 +63,31 @@ const ThemelessAudioPlayerPlaybackSpeedControl = React.forwardRef<
...
-        </IconButton>
+        {props.children ? (
+          // If there are children, trigger the open state
+          // on click and pass the original onClick handler
+          React.cloneElement(props.children, {
+            onClick: () => {
+              setIsOpen(open => !open);
+              if (
+                props.children &&
+                typeof props.children.props.onClick === 'function'
```
Comment (mstuartf) [inline src/audio-player-composable/components/playback-speed-control/playback-speed-control.tsx:74]: Not important but it's probably sufficient to just check if `onClick` is defined. If so, we can probably assume it's a function.

File: src/audio-player-composable/components/playback-speed-control/playback-speed-control.tsx [hunk 2]
```diff
@@ -63,15 +63,31 @@ const ThemelessAudioPlayerPlaybackSpeedControl = React.forwardRef<
...
-          size={buttonSize}
-        >
-          <IconFilledSlowMotionVideo />
-        </IconButton>
+        {props.children ? (
+          // If there are children, trigger the open state
+          // on click and pass the original onClick handler
+          React.cloneElement(props.children, {
+            onClick: () => {
+              setIsOpen(open => !open);
```
Comment (mstuartf) [inline src/audio-player-composable/components/playback-speed-control/playback-speed-control.tsx:71]: You could also define a function for this callback and use it below too. Again not important so feel free to ignore.

File: src/audio-player-composable/types.ts [hunk 3]
```diff
@@ -46,6 +46,7 @@ export interface AudioPlayerProviderContext {
   loading: boolean;
   audioRef: React.RefObject<HTMLAudioElement>;
   audioSectionRef: React.RefObject<HTMLDivElement>;
+  playbackSpeed: number;
```
Comment (mstuartf) [inline src/audio-player-composable/types.ts:49]: Why was this added here? I can't quite see.

File: src/audio-player-composable/components/playback-speed-control/types.ts [hunk 4]
```diff
@@ -17,6 +17,7 @@ export interface AudioPlayerPlaybackSpeedControlOverridesProps {
 }
 
 export interface AudioPlayerPlaybackSpeedControlProps {
+  children?: React.ReactElement;
```
Comment (mstuartf) [inline src/audio-player-composable/components/playback-speed-control/types.ts:20]: Do we want to support giving users access to the audio player state in their custom component? E.g. they might want to show the current playback speed on their button. We could use render props for this. Or perhaps they can already do this using context. If so, ignore me.

File: src/audio-player-composable/components/playback-speed-control/playback-speed-control.tsx (hunk 1 above)
Comment (evgenitsn) [reply src/audio-player-composable/components/playback-speed-control/playback-speed-control.tsx]: Updated

File: src/audio-player-composable/types.ts (hunk 3 above)
Comment (evgenitsn) [reply src/audio-player-composable/types.ts]: Actually we don't need this cause we can get it from `getPlaybackSpeedControlProps`. Removed.

File: src/audio-player-composable/components/playback-speed-control/types.ts (hunk 4 above)
Comment (evgenitsn) [reply src/audio-player-composable/components/playback-speed-control/types.ts:20]: We are already doing that using Context yes. 

You can have a look at this storybook scenario: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2314-audio-player-playback-speed-custom-children-trigger/storybook/?path=/story/newskit-light-audio-player-composable--audio-player-playback-speed-trigger-button

Comment (mstuartf) [reply src/audio-player-composable/components/playback-speed-control/types.ts:20]: Ah yes, thanks.

**Review by mstuartf — DISMISSED** [review]

**Review by mutebg — DISMISSED** [review]

**Comment by evgenitsn** [comment]
@mutebg @mstuartf the last merge from `main` marked the current reviews as **stale**, please reapprove. Thanks!

**Review by mstuartf — APPROVED** [review]

**Review by mutebg — APPROVED** [review]

**Pull request merged**